docs/source/reference/storage.rst

```diff
@@ -1,201 +1,131 @@
+the task can access these objects as if they were local files.
 
-A storage object can used in either :code:`MOUNT` mode or :code:`COPY` mode.
 
-* In :code:`MOUNT` mode, the backing store is directly "mounted" to the remote VM.
-  I.e., files are streamed when accessed by the task and all writes are replicated
-  to remote bucket (and any other VMs mounting the same bucket).
 
-* In :code:`COPY` mode, the files are pre-fetched and cached on the local disk.
-  Writes are not replicated on the remote store.
+Usage
+-----
 
-.. note::
-    sky.Storage does not guarantee preservation of file
-    permissions - you may need to set file permissions during task execution.
+Object stores are specified using the :code:`file_mounts` field in a SkyPilot task.
 
-Using SkyPilot Storage
-----------------------
-SkyPilot Storage can be used by specifying additional fields in the
-:code:`file_mounts`. By default, :code:`file_mounts` uses rsync to
-directly copy files from local to remote VM.
-However, you can have them backed by SkyPilot Storage, which uploads
-the files to a cloud store (e.g. S3, GCS, R2, or IBM) and have them persist there by
-specifying the :code:`name`, :code:`source` and :code:`persistent` fields. By
-enabling persistence, file_mount sync can be made significantly faster.
+.. tab-set::
 
-Here is an example of a :code:`file_mount` that uses SkyPilot Storage:
+    .. tab-item:: Use existing bucket
+        :sync: existing-bucket-tab
 
-.. code-block:: yaml
+        To access a bucket created externally (e.g., through cloud CLI or other tools),
+        specify ``source``.
 
-    file_mounts:
-      /mybucket:
-        name: my-sky-bucket # Make sure it is unique or you own this bucket name
-        source: ~/dataset # Contents of the store. Can be local or an object store path.
-        store: s3 # Could be either of [s3, gcs, r2]. Defaults to None.
-        persistent: True  # Set to False to delete the bucket after the task is done. Defaults to True.
-        mode: MOUNT  # MOUNT or COPY. Defaults to MOUNT if not specified
+        .. code-block:: yaml
 
+          # Mount an existing S3 bucket
+          file_mounts:
+            /mydata:
+                source: s3://my-bucket/ # or gs://, r2://, cos://<region>/<bucket>
+                mode: MOUNT  # Optional - either MOUNT or COPY. Defaults to MOUNT.
 
-Common Use Cases
-^^^^^^^^^^^^^^^^
+        This will `mount <storage-mounting-modes_>`__ the contents of the bucket at ``s3://my-bucket/`` to the remote VM at ``/mydata``.
 
-1.  **You want to upload your local data to remote VM -** specify the name and
-    source fields. Name sets the bucket name that will be used, and source
-    specifies the local path to be uploaded.
 
-2.  **You want to mount an existing S3/GCS/R2 bucket to your remote VM -** specify
-    just the source field (e.g., s3://my-bucket/, gs://my-bucket/ or r2://my-bucket/).
+    .. tab-item:: Upload files to new bucket
```

Comment:
Does this mean we are not able to upload files to existing bucket? This might be unexpected, as in some of our users use case, they woud like to have all the data in a single bucket.

Reply:
I've reworded the titles and added a visually distinct`note` - does that help?

Reply:
Ahh, I was wondering what will happen if the bucket already exists and was not created by SkyPilot. For example, there is a team writing files to the same bucket, and the bucket will only be `managed by SkyPilot` for one of the teammates.

Reply:
Just a kind bump for this @romilbhardwaj : )

Reply:
Ahh thanks for the ping @Michaelvll, this had slipped through the cracks.

If the bucket was created by a team member, other members will have to use the `source: s3://...` pattern. 

We disallow `name: external-bucket` pattern from mounting external buckets because e.g., if I want to create a bucket and use the name of a pre-existing public (or private) bucket, these semantics will silently mount the bucket and my task will fail when it tries to write the desired path (or worse yet, overwrite unintended files).

This behavior manifested in our smoke tests too (and is now fixed in #2804). Our `test_using_file_mounts_with_env_vars` was silently re-mounting `skypilot-temp-gcs-test` bucket which was shared in our GCS account, instead of creating a new bucket at each test instance.

Reply:
Thanks that makes sense! I found it a bit non-intuitive, for the following case, but I guess it is fine to leave it to the future.

If I am a member of a org, and created a bucket for with mount mode for storing some jupyter notebooks:
```
file_mounts:
  /dst:
    name: my-bucket
    store: s3
    mode: MOUNT
run: jupyter notebook /dst
```
The person then shares the YAML with another member in the org. Does this mean the other member running this yaml will fail and have to change it to:
```
file_mounts:
  /dst:
    source: s3://my-bucket
    mode: MOUNT
run: jupyter notebook /dst
```